### PR TITLE
core[minor]: Add Runnable.batch_as_completed

### PR DESCRIPTION
**Pull request by nfcampos**
This PR adds `batch as completed` method to the standard Runnable interface. It takes in a list of inputs and yields the corresponding outputs as the inputs are completed.

### CODE REVIEW

**Review by eyurtsev — APPROVED** [review]

File: libs/core/langchain_core/runnables/base.py [hunk 1]
```diff
@@ -567,6 +613,40 @@ async def ainvoke(
...
         return await gather_with_concurrency(configs[0].get("max_concurrency"), *coros)
 
+    async def abatch_as_completed(
+        self,
+        inputs: List[Input],
+        config: Optional[Union[RunnableConfig, List[RunnableConfig]]] = None,
+        *,
+        return_exceptions: bool = False,
+        **kwargs: Optional[Any],
+    ) -> AsyncIterator[Tuple[Input, Output]]:
```
Comment (eyurtsev) [inline libs/core/langchain_core/runnables/base.py:623]: Do we need this to be `AsyncIterator[Tuple[int, Input, Output]]`?

Duplicated Input, Output but different configs

Comment (nfcampos) [reply libs/core/langchain_core/runnables/base.py:623]: like index in input array?

Comment (nfcampos) [reply libs/core/langchain_core/runnables/base.py:623]: in that case I'd actually prefer to not return the input at all, just index and output, wdyt?

Comment (eyurtsev) [reply libs/core/langchain_core/runnables/base.py:623]: yeah that works

**Review by baskaryan — APPROVED** [review]

**Pull request merged**